control/statesp.py

```diff
@@ -1215,17 +1216,16 @@ def append(self, other):
 
     def __getitem__(self, indices):
         """Array style access"""
-        if len(indices) != 2:
+        if not isinstance(indices, Iterable) or len(indices) != 2:
             raise IOError('must provide indices of length 2 for state space')
-        outdx = indices[0] if isinstance(indices[0], list) else [indices[0]]
-        inpdx = indices[1] if isinstance(indices[1], list) else [indices[1]]
+        outdx, inpdx = indices
+        if not isinstance(outdx, (int, slice)) or not isinstance(inpdx, (int, slice)):
+            raise TypeError(f"system indices must be integers or slices")
         sysname = config.defaults['iosys.indexed_system_name_prefix'] + \
             self.name + config.defaults['iosys.indexed_system_name_suffix']
         return StateSpace(
             self.A, self.B[:, inpdx], self.C[outdx, :], self.D[outdx, inpdx],
-            self.dt, name=sysname,
-            inputs=[self.input_labels[i] for i in list(inpdx)],
-            outputs=[self.output_labels[i] for i in list(outdx)])
+            self.dt, name=sysname, inputs=self.input_labels[inpdx], outputs=self.output_labels[outdx])
```

Comment:
Was there a reason for this change?  It looks like the updated final line goes past column 79, which we try not to do.

Reply:
The change was needed since `slice` is not iterable. I have pushed a new commit with a maximum column length of 79.